### ISSUE TITLE
"docker rm -f" cannot remove pausing container.

### ISSUE THREAD

**Issue by yoshiokatsuneo** [moby/moby#28366]
**Description**

"docker rm -f" does not remove the container but cause error:

```
Error response from daemon: Could not kill running container 0859eb46e6f5e76161b06f12678bd95c1b55dde8c6f4645ca660faa6da201450, cannot remove - Container 0859eb46e6f5e76161b06f12678bd95c1b55dde8c6f4645ca660faa6da201450 is paused. Unpause the container before stopping
```

Yes, I can "unpause" before killing the container. But, it is quite nice if "docker rm -f" is the final command to remove the container. Also, strictly speaking, even if I call "docker unpause" just before "docker rm -f", Someone may "unpause" the container between those commands.

**Steps to reproduce the issue:**
1. Run a container.

```
# docker run -d ubuntu sleep 1000
acf0571d8fdc581a61428b7e9312b12f378d4d649a55cd6e0c199c597a53596e
```

2. Pause the container
```
# docker pause acf0571d8fdc581a61428b7e9312b12f378d4d649a55cd6e0c199c597a53596e
acf0571d8fdc581a61428b7e9312b12f378d4d649a55cd6e0c199c597a53596e
```

3. Remove the container
```
# docker rm -f acf0571d8fdc581a61428b7e9312b12f378d4d649a55cd6e0c199c597a53596e
Error response from daemon: Could not kill running container acf0571d8fdc581a61428b7e9312b12f378d4d649a55cd6e0c199c597a53596e, cannot remove - Container acf0571d8fdc581a61428b7e9312b12f378d4d649a55cd6e0c199c597a53596e is paused. Unpause the container before stopping
```

**Describe the results you received:**
Error message:

```
Error response from daemon: Could not kill running container acf0571d8fdc581a61428b7e9312b12f378d4d649a55cd6e0c199c597a53596e, cannot remove - Container acf0571d8fdc581a61428b7e9312b12f378d4d649a55cd6e0c199c597a53596e is paused. Unpause the container before stopping
```

**Describe the results you expected:**
Removing the container

**Output of `docker version`:**

```
Client:
 Version:      1.12.2
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   bb80604
 Built:        Tue Oct 11 18:29:41 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.2
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   bb80604
 Built:        Tue Oct 11 18:29:41 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 16
 Running: 13
 Paused: 1
 Stopped: 2
Images: 4
Server Version: 1.12.2
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 150
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-43-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.858 GiB
Name: paizaterm-prod4
ID: 757W:AQM2:WNUV:CQSJ:URNH:5I5W:65OW:4DXQ:QYLV:M57U:P5AG:WBAV
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Labels:
 provider=generic
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
AWS

**Comment by coolljt0725** [comment]
I think we should implement killing a paused container directly without unpause it first, because if a container is paused by intentionally and I want kill it, but unpause it will cause problem.

**Comment by allencloud** [comment]
I also found that the error message is not so readable: 
```
Error response from daemon: Could not kill running container acf057..., cannot remove - Container acf057... is paused. Unpause the container before stopping
```

**Comment by allencloud** [comment]
I am on @coolljt0725 's side. Unpausing a container first before killing it may be an improper choice.
While, if we change the original killing logic, I think it will break back compatibility of action `docker kill`. So maybe add a brand new function for killing paused container specific for `docker rm -f ` may work for us.

Any input? @coolljt0725 @thaJeztah @vdemeester

**Comment by thaJeztah** [comment]
Making `docker rm -f` "just work" sounds good to me; do we know what's needed to make it work for a paused container?

**Comment by allencloud** [comment]
I tested just now, and I found that we could not use `kill -9 container_init_pid` to kill a paused container.(If not correct, please correct me.) So maybe unpausing it first is a good choice.  Here are two ways:
1. only make `docker rm -f paused_container` unpause the container, this way will not influence `docker kill paused_container`, in another word `docker kill paused_container` still returns an error;
2. we change the logic of `func (daemon *Daemon) killWithSignal` of https://github.com/docker/docker/blob/master/daemon/kill.go#L57, however this will influence `docker kill paused_container`, and this will affect the back compatibility.

**Comment by coolljt0725** [comment]
relate to https://github.com/docker/docker/pull/13304 https://github.com/docker/docker/issues/15853

**Comment by thaJeztah** [comment]
The PR implementing this (https://github.com/moby/moby/issues/34027) will be included in the upcoming Docker 17.07 release
